lib/srv/authhandlers.go

```diff
@@ -521,14 +553,35 @@ func (h *AuthHandlers) IsHostAuthority(cert ssh.PublicKey, address string) bool
 	return true
 }
 
+// loginChecker checks if the Teleport user should be able to login to
+// a node. It is created for tests.
+type loginChecker interface {
+	// canLoginWithoutRBAC checks the given certificate (supplied by a connected
+	// client) to see if this certificate can be allowed to login as user:login
+	// pair to requested server.
+	canLoginWithoutRBAC(cert *ssh.Certificate, clusterName string, teleportUser string, osUser string) error
+	// canLoginWithRBAC checks the given certificate (supplied by a connected
+	// client) to see if this certificate can be allowed to login as user:login
+	// pair to requested server and if RBAC rules allow login.
+	canLoginWithRBAC(cert *ssh.Certificate, clusterName string, node types.Server, teleportUser, osUser string) error
+}
+
+type ahLoginChecker struct {
+	h *AuthHandlers
+}
+
 // canLoginWithoutRBAC checks the given certificate (supplied by a connected
 // client) to see if this certificate can be allowed to login as user:login
 // pair to requested server.
-func (h *AuthHandlers) canLoginWithoutRBAC(cert *ssh.Certificate, clusterName string, teleportUser, osUser string) error {
-	h.log.Debugf("Checking permissions for (%v,%v) to login to node without RBAC checks.", teleportUser, osUser)
+func (a *ahLoginChecker) canLoginWithoutRBAC(cert *ssh.Certificate, clusterName string, teleportUser, osUser string) error {
+	if a.h.c.AuthorizeWithoutRBAC != nil {
+		return a.h.c.AuthorizeWithoutRBAC(cert, clusterName, teleportUser, osUser)
+	}
```

Comment:
As we talked about earlier, please remove this redirection. It's not needed anymore since the `AuthHandler` can accept different implementations of `loginChecker` interface.

Reply:
Done